xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/DefaultCompositeEventStatusManager.java

```diff
@@ -46,39 +47,47 @@
 @Singleton
 public class DefaultCompositeEventStatusManager implements CompositeEventStatusManager
 {
+    private static final int BATCH_SIZE = 50;
+
     @Inject
     private EventStore eventStore;
 
     @Override
     public List<CompositeEventStatus> getCompositeEventStatuses(List<CompositeEvent> compositeEvents, String entityId)
```

Comment:
Ideally, the real solution would be to stop needed this method at all...

Reply:
FTR: https://jira.xwiki.org/browse/XWIKI-21319